src/cattr/converters.py

```diff
@@ -194,8 +194,9 @@ def unstructure_attrs_asdict(self, obj):
         rv = self._dict_factory()
         for a in attrs:
             name = a.name
-            v = getattr(obj, name)
-            rv[name] = dispatch(v.__class__)(v)
+            if hasattr(obj, name):
+                v = getattr(obj, name)
+                rv[name] = dispatch(v.__class__)(v)
```

Comment:
This will only unstructure attributes that are set on the instance. This works when the `init=False` attributes are not set at unstructure time, but how does it work if they are? If the attribute is set, won't the structuring code attempt to initialise it? (See `structure_attrs_fromdict`.) I think we need to explicitly check if an attribute has `init` set to true, and exclude ones that don't, whether they are actually set on the instance or not.

Additionally, this behaviour needs to be added to `unstructure_attrs_astuple` below.

Reply:
The code only deal with un-structure, if the attribute is set, the attribute will be unstructured, which is the default behavior.